torch_agents/cogment_verse_torch_agents/selfplay_td3/selfplay_td3.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2021 AI Redefined Inc. <dev+cogment@ai-r.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cogment_verse_torch_agents.selfplay_td3.replaybuffer import Memory
+from cogment_verse_torch_agents.selfplay_td3.model import ActorNetwork, CriticNetwork
+import numpy as np
+import torch.nn.functional as F
+import torch
+import copy
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+# pylint: disable=C0103
+# pylint: disable=W0613
+# pylint: disable=W0221
+# pylint: disable=W0212
+# pylint: disable=W0622
+class SelfPlayTD3:
```

Comment:
The name is a bit confusing. This class is just a TD3 and doesn't deal with selfplay yet ?

Reply:
Unlike a regular TD3, this implementation takes into account "alice" and "bob" agent, hence the name.